intern/cycles/kernel/kernel_types.h

```diff
@@ -506,30 +509,30 @@ typedef ccl_addr_space struct PathRadiance {
 #endif
 
   float transparent;
-  float3 emission;
+  SceneLinearColor emission;
 #ifdef __PASSES__
-  float3 background;
-  float3 ao;
+  SceneLinearColor background;
```

Comment:
This `SceneLinearColor` (and the similar `SpectralColor`) are just an alias for float3, but I've used this so that it is clear what the float3 represents in different parts of the engine. A little tool that has helped me massively to not get lost in the complexity of a specific function. Adding the annotations to the kernel isn't yet complete, but I've just been converting them whenever I see them.

I _could_ have a new class/type/struct for SpectralColor, and I do plan to do that in the future, but if I did that now, nothing would compile until everything was updated, and my last 2 attempts at a 'big bang' conversion failed in me running out of motivation and being too confused by the complexity of a particular area. This approach has allowed me to make incremental changes and see something come out the other end.